opencti-platform/opencti-graphql/src/database/file-storage.js

```diff
@@ -237,25 +237,91 @@ export const storeFileConverter = (user, file) => {
 };
 
 /**
- * Get S3 metadata information of file like size (calling HEAD on S3 file).
+ * Get file size from S3 (calling HEAD on S3 file).
  */
-export const loadFile = async (user, fileS3Path, opts = {}) => {
-  const { dontThrow = false } = opts;
+export const getFileSize = async (user, fileS3Path) => {
   try {
     const object = await s3Client.send(new s3.HeadObjectCommand({
       Bucket: bucketName,
       Key: fileS3Path
     }));
-    const mimeTypeResolved = guessMimeType(fileS3Path);
+    return object.ContentLength;
+  } catch (err) {
+    throw UnsupportedError('Load file from storage fail', { cause: err, user_id: user.id, filename: fileS3Path });
+  }
+};
+
+/**
+ * Get file metadata from database, or else from S3.
+ */
+export const loadFile = async (context, user, fileS3Path, opts = {}) => {
+  const fileInformationFromDB = await documentFindById(context, user, fileS3Path);
+  const { dontThrow = false } = opts;
+  try {
+    let metaData; let name; let size; let lastModified; let lastModifiedSinceMin;
+
+    // Try first to get metadata from elastic
+    if (fileInformationFromDB) {
+      const dbMeta = fileInformationFromDB.metaData;
+      metaData = {
+        version: dbMeta.version,
+        // description: object.Metadata.description,
+        // list_filters: object.Metadata.list_filters,
+        filename: dbMeta.filename,
+        mimetype: dbMeta.mimetype,
+        // labels_text: object.Metadata.labels_text,
+        // labels: object.Metadata.labels_text ? object.Metadata.labels_text.split(';') : [],
+        encoding: dbMeta.encoding,
+        creator_id: dbMeta.creator_id,
+        entity_id: dbMeta.entity_id,
+        // external_reference_id: object.Metadata.external_reference_id,
+        // messages: object.Metadata.messages,
+        // errors: object.Metadata.errors,
+        inCarousel: dbMeta.inCarousel,
+        order: dbMeta.order
+      };
+      name = fileInformationFromDB.name;
+      size = fileInformationFromDB.size;
+      lastModified = fileInformationFromDB.lastModified;
+      lastModifiedSinceMin = lastModified; // ??
+    } else {
+      // Else try to get them from S3 instead
+      const object = await s3Client.send(new s3.HeadObjectCommand({
+        Bucket: bucketName,
+        Key: fileS3Path
+      }));
+      metaData = {
+        version: object.Metadata.version,
```

Comment:
not sure this would work, since we don't store Metadata anymore in S3, or is it just a fallback for previously stored files ? I wonder if object.Metadata would be defined for new files.